### ISSUE TITLE
Can't spawn up multiple connections

### ISSUE THREAD

**Issue by Den3** [lucacorti/lapin#23]
This is my configuration
```
config :lapin, :connections, [
  [
    module: consumer,
    uri: System.get_env("AMQP_URL"),
    channels: [
      [
        role: :consumer,
        pattern: Lapin.Pattern.Config,
        exchange: exchange,
        queue: queue,
        routing_key: queue,
        exchange_type: :topic,
        consumer_ack: true,
        consumer_prefetch: 1,
        queue_arguments: [
          {"x-dead-letter-exchange", exchange_dlx},
          {"x-dead-letter-routing-key", queue_dlx}
        ]
      ],
      [
        role: :producer,
        pattern: Lapin.Pattern.Config,
        exchange: exchange,
        queue: queue,
        routing_key: queue,
        exchange_type: :topic,
        consumer_ack: true,
        consumer_prefetch: 1,
        queue_arguments: [
          {"x-dead-letter-exchange", exchange_dlx},
          {"x-dead-letter-routing-key", queue_dlx}
        ]
      ]
    ]
  ],
  [
    module: another_consumer,
    uri: System.get_env("AMQP_URL"),
    channels: [
      [
        role: :consumer,
        pattern: Lapin.Pattern.Config,
        exchange: exchange,
        queue: queue,
        routing_key: queue,
        exchange_type: :topic,
        consumer_ack: true,
        consumer_prefetch: 1,
        queue_arguments: [
          {"x-dead-letter-exchange", exchange_dlx},
          {"x-dead-letter-routing-key", queue_dlx}
        ]
      ]
    ]
  ]
]
```

This one comes with the error `duplicated id Lapin.Connection found in the supervisor specification, please explicitly pass the :id option when defining this worker/supervisor`.

After some digging in, I found the issue may be [this](https://github.com/lucacorti/lapin/blob/master/lib/lapin/connection/supervisor.ex#L17). It should provide id for each worker

**Comment by lucacorti** [comment]
This should not happen unless you specify the same `module` in the connection configuration as connections are registered by module.

**Comment by Den3** [comment]
Hi @lucacorti . I reproduce the error in this [repo](https://github.com/Den3/lapin_bug). My fix is [this](https://github.com/Den3/lapin/commit/6ee2e1e2c97b3dc1d89b89250dea7f7b5e2dfed2) . Can you have a look to see if my config is wrong? Thank you

**Comment by Den3** [comment]
I've created a [PR](https://github.com/lucacorti/lapin/pull/24). Please review. Thank you.

**Comment by lucacorti** [comment]
Fixed in `0.3.4`. Thanks.
